infrastructure/main.tf

```diff
@@ -36,37 +36,83 @@ locals {
   }
 }
 
+data "google_compute_network" "existing-network" {
```

Comment:
lets keep existing network PSA enable out for now..

because now we dont know if PSA is already enabled for exisiting network and that will lead to apply failures.

We do PSA for all new network only on create_network=true

Reply:
This data stanza isn't enabling PSA, it's just a data reference to an existing resource. However it's only being used to reference "network_self_link", which is only used in the PSA generation (when create_network=true). So I think I can follow the same pattern as "cloud-nat" module below, and directly reference module.custom-network[0], rather than a local variable that is conditionally set. That will allow me to remove this data stanza.

Reply:
Actually, I'll keep this stanza. Having the `data` block in here validates that the existing network actually exists during `terraform apply` (by returning an error if it does not exist). This gives the user immediate feedback if they have misconfigured their Terraform.